LMakshow/virtual-keyboard
# Virtual Keyboard task

1. Task: https://github.com/rolling-scopes-school/tasks/blob/master/tasks/virtual-keyboard/virtual-keyboard-en.md
2. App screenshot: ![Screenshot_15](https://user-images.githubusercontent.com/19377176/166896096-a7932e29-c25b-40b0-bfb8-f0712b96d135.jpg)
3. Deploy:  https://lmakshow.github.io/virtual-keyboard.
4. How to test: 
- Check [repository](https://github.com/LMakshow/virtual-keyboard/tree/virtual-keyboard-dev) and [deploy](https://lmakshow.github.io/virtual-keyboard).
- To prevent CRLF repo clone error use in terminal `git config --global core.autocrlf input`
- Clone repo and check `virtual-keyboard-dev` branch.
- Check eslint errors using `npm install eslint --save-dev` and `npx eslint src --ext js`.
5. Done 05.05.22 (deadline 10.05.22)
6. Score: 110 / 110
- [x] Minimal scope
    - [x] the generation of DOM elements is implemented. body in the index.html is empty (can contain only script tag): +20 
    - [x] pressing a key on a physical keyboard highlights the key on the virtual keyboard (numbers, letters, punctuation marks, backspace, del, enter, shift, alt, ctrl, tab, caps lock, space, arrow keys): +10
- [x] Basic scope
    - [x] switching keyboard layouts between English and Russian is implemented. Selected language saved and used on page reload. A keyboard shortcut for switching a language is Shift+Alt: +15
    - [x] mouse clicks on buttons of the virtual keyboard or pressing buttons on a physical keyboard inputs characters to the input field (text area): +15
- [x] Extra scope
    - [x] animation of pressing a key is implemented: +15
- [x] Technical requirements
    - [x] usage of ES6+ features (class Keyboard, import, arrow functions, ternary operators, property destructuring, etc): +15
    - [x] usage of ESLint: +10
    - [x] Requirements to the repository, commits and pull request are met: +10
- [x] Penalties
    - [x] No executable code errors or eslint-config-airbnb-base warnings 
7. **Замечание:** использовалось правило import/extensions в .eslintrc.js, по согласию с куратором просьба не снижать за это баллы.
